homeassistant/components/arve/sensor.py

```diff
@@ -0,0 +1,101 @@
+"""Sensor platform for Arve devices."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from asyncarve import ArveSensProData
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    CONCENTRATION_PARTS_PER_MILLION,
+    PERCENTAGE,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import ArveCoordinator
+from .entity import ArveDeviceEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class ArveDeviceEntityDescription(SensorEntityDescription):
+    """Describes Arve device entity."""
+
+    value_fn: Callable[[ArveSensProData, str], float | int] = (
+        lambda arve_data, key: getattr(arve_data, key.lower())
+    )
+
+
+SENSORS: tuple[ArveDeviceEntityDescription, ...] = (
+    ArveDeviceEntityDescription(
+        key="CO2",
+        translation_key="co2",
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        device_class=SensorDeviceClass.CO2,
+    ),
+    ArveDeviceEntityDescription(
+        key="AQI",
+        translation_key="aqi",
+        native_unit_of_measurement=None,
+        device_class=SensorDeviceClass.AQI,
+    ),
+    ArveDeviceEntityDescription(
+        key="Humidity",
+        translation_key="humidity",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.HUMIDITY,
+    ),
+    ArveDeviceEntityDescription(
+        key="PM10",
+        translation_key="pm10",
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        device_class=SensorDeviceClass.PM10,
+    ),
+    ArveDeviceEntityDescription(
+        key="PM25",
+        translation_key="pm25",
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        device_class=SensorDeviceClass.PM25,
+    ),
+    ArveDeviceEntityDescription(
+        key="Temperature",
+        translation_key="temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+    ),
```

Comment:
I don't know why these have a translation key. The device class already takes care of both the icon and the name of the entity